File: Coin Change-2.py

class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        
        if amount == 0:
            return 1
        result = [0]*(amount+1)
        for i in range(len(coins)):
            for j in range(1,amount+1):
                if j < coins[i]:
                    result[j] = result[j]
                elif j == coins[i]:
                    result[j] = result[j]+1
                else:
                    result[j] = result[j] + result[j-coins[i]]
        
        return result[amount]

File: Coin_Change_2.cs
public class Solution
{
    public int Change(int amount, int[] coins)
    {
        int m = coins.Length;
        int n = amount;
        int[][] dp = new int[m + 1][n + 1];

        dp[0][0] = 1;
        for (int i = 1; i <= m; i++)
        {
            for (int j = 0; j <= n; j++)
            {
                if (j < coins[i - 1])
                {
                    dp[i][j] = dp[i - 1][j];
                }
                else
                {
                    dp[i][j] = dp[i - 1][j] + dp[i][j - coins[i - 1]];
                }
            }
        }
        return dp[m][n];
    }
}